### PR TITLE
migrate to bnd

### PR DESCRIPTION
**Pull request by J-N-K**
Signed-off-by: Jan N. Klug <jan.n.klug@rub.de>

### CODE REVIEW

**Comment by J-N-K** [comment]
Please check if this is working!

**Comment by J-N-K** [comment]
Interesting. It builds locally. I‘ll check tomorrow.

**Comment by J-N-K** [comment]
Some comments:

- SAT shows a lot of errors, I guess that should be fixed. 
- There are some usages of guava, shall we replace those?
- I'll try to figure out why I have to add the parent-pom and remove it again.
- Can someone test if this works as expected?

**Comment by openhab-5iver** [comment]
@J-N-K, I can do a build, but it would save time if you could post a jar to test with.

**Comment by J-N-K** [comment]
Try [this](https://janessa.me/esh/org.openhab.binding.zwave-bnd.jar).

**Comment by openhab-5iver** [comment]
Thank you! I've had it running for about 30 minutes and all looks good so far. No errors and devices are functioning. I had my fingers crossed that migrating to BND might [magically solve this issue](https://community.openhab.org/t/no-serialport-provider-error-with-ubuntu-18-04-and-zwave-2-5-x-snapshot/72927/16?u=5iver), but it did not (likely USB discovery problem). I'll report back if I find any issues!

Would you be able to migrate the Zigbee binding too?

**Comment by J-N-K** [comment]
I‘m not sure where the feature for this Binding is defined. That needs changes, too. 

Zigbee should work the same, so I‘ll have a look at that, too.

**Comment by openhab-5iver** [comment]
> I‘m not sure where the feature for this Binding is defined. That needs changes, too.

Are you looking for this?
https://github.com/openhab/openhab-distro/blob/master/features/addons/src/main/feature/feature.xml

**Comment by J-N-K** [comment]
I removed alldependencies to Guava, so we don't have to add that dependency any longer. There are still a lot of SAT errors (many often related to wrong file headers). That shopuld probably be cleaned up in a follow-up PR. I have prepared a new download, so if anyone wants to check: The link above is updated.

I'll prepare the PR for the feature, it should be merged immediately after this one.

File: src/main/java/org/openhab/binding/zwave/event/BindingEventFactory.java [hunk 1]
```diff
@@ -46,7 +46,7 @@ public BindingEventFactory(Set<String> supportedEventTypes) {
     }
 
     public BindingEventFactory() {
-        super(Sets.newHashSet(BindingEvent.TYPE));
+        super(Stream.of(BindingEvent.TYPE).collect(Collectors.toSet()));
```
Comment (cweitkamp) [inline src/main/java/org/openhab/binding/zwave/event/BindingEventFactory.java]: Minor tweak: `Collections.singleton()`?

Comment (J-N-K) [reply src/main/java/org/openhab/binding/zwave/event/BindingEventFactory.java]: ... of course.

**Comment by cdjackson** [comment]
@J-N-K sorry I didn't notice that this moved all XML files and there is now a conflict. Can you rebase this and then I'll merge. It's pretty difficult to review with 1000 files showing as changed ;)

**Comment by J-N-K** [comment]
done

**Comment by cdjackson** [comment]
Thanks.

**Pull request merged**